tests/main/snap-quota-thread/task.yaml

```diff
@@ -0,0 +1,118 @@
+  snap start test-snapd-stressd.spawner
+  sleep 1
+
+  echo "The systemd slice should be active now"
+  sliceName="snap.$(systemd-escape --path group-one).slice"
+  systemctl show --property=ActiveState "$sliceName" | MATCH "ActiveState=active"
+
+  echo "The service should be active"
+  snap services test-snapd-stressd.spawner | MATCH "test-snapd-stressd.spawner\s+disabled\s+active"
+
+  # systemd/kernel have three different locations for the cgroup pids depending
+  # on version
+  echo "The systemd slice should have one process in it now"
+  cgroupsV1OldSystemdProcsFile="/sys/fs/cgroup/pids/$sliceName/snap.test-snapd-stressd.spawner.service/cgroup.procs"
+  cgroupsV1ProcsFile="/sys/fs/cgroup/pids/$sliceName/cgroup.procs"
+  cgroupsV2ProcsFile="/sys/fs/cgroup/$sliceName/snap.test-snapd-stressd.spawner.service/cgroup.procs"
+  if [ -e "$cgroupsV2ProcsFile" ]; then
+      cgroupProcsFile="$cgroupsV2ProcsFile"
+  elif [ -e "$cgroupsV1OldSystemdProcsFile" ]; then
+      cgroupProcsFile="$cgroupsV1OldSystemdProcsFile"
+  elif [ -e "$cgroupsV1ProcsFile" ]; then
+      cgroupProcsFile="$cgroupsV1ProcsFile"
+  else
+      echo "cannot detect cgroup procs file"
+      exit 1
+  fi
+
+  #shellcheck disable=SC2016
+  retry --wait 1 -n 100 --env cgroupProcsFile="$cgroupProcsFile" sh -x -c 'test "$(wc -l < $cgroupProcsFile)" = 1'
+  SERVER_PID=$(cat "$cgroupProcsFile")
+
+  echo "And that process is the main PID for spawner"
+  systemctl show --property=MainPID snap.test-snapd-stressd.spawner.service | MATCH "MainPID=$SERVER_PID"
+
+  echo "And the service is in the Control Group for the slice"
+  # using a regexp for the ControlGroup match here as on older systemd (16.04)
+  # the group is double escaped
+  systemctl show --property=ControlGroup snap.test-snapd-stressd.spawner.service | MATCH 'ControlGroup=/snap.group(.*)one.slice/snap.test-snapd-stressd.spawner.service'
+
+  # snap quota group-one formats the quota usage as a nice human readable 
+  # string, which complicates the comparison here, so instead just grab the 
+  # task usage from the REST API instead
+  echo "Verify the number of threads is correct (=16)"
+  
+  threadUsage="$(sudo snap run test-snapd-curl.curl --unix-socket /run/snapd.socket http://localhost/v2/quotas/group-one | jq -r '.result.current.threads')"
+  if ! [ "$threadUsage" -eq 16 ]; then
+    echo "thread usage reported does not equal the expected quota usage $threadUsage"
+    exit 1
+  fi
+
+  echo "Increase the thread quota to 24"
+  snap set-quota group-one --thread=24
+
+  echo "Verify the number of threads is correct (=24)"
+  sleep 3 # sleep the delay time (2s) and one more to allow for threads to start
+  threadUsage="$(sudo snap run test-snapd-curl.curl --unix-socket /run/snapd.socket http://localhost/v2/quotas/group-one | jq -r '.result.current.threads')"
+  if ! [ "$threadUsage" -eq 24 ]; then
+    echo "thread usage reported does not equal the expected quota usage $threadUsage"
+    exit 1
+  fi
```

Comment:
Should we also use "snap quota" here to see the current task from the commandline ? Also do we need to test the api explicitly given that the commandline already hits it?